monkey/monkey_island/cc/resources/utils/semaphores.py

```diff
@@ -0,0 +1,5 @@
+from gevent.lock import BoundedSemaphore
+
+# Semaphore avoids race condition between monkeys
+# being marked dead and monkey waking up as alive
+AGENT_KILLING_SEMAPHORE = BoundedSemaphore()
```

Comment:
Why a semaphore instead of a lock? If the code paths that are being controlled are mutually exclusive, it's usually preferable to use a lock.

Reply:
I suppose gevent doesn't offer a lock, because a semaphore of 1 is basically a lock

Reply:
Ah, I missed that it was a gevent object. Perhaps name it `..._mutex` instead of `semaphore`, since this more accurately describes its purpose. Also, I wouldn't make it all capitals, it's not a constant.